eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -330,69 +311,37 @@ ratifyTransition = do
         && spoAccepted env st gas
         && dRepAccepted env st gas
         then do
-          let siblings = getSiblings ensPrevGovActionIdsChildren gasAction
-          -- Update ENACT state with the governance action that was ratified
-          es <-
+          newEnactState <-
             trans @(EraRule "ENACT" era) $
               TRC ((), rsEnactState, EnactSignal gasId gasAction)
-          let st' =
-                st
-                  { rsEnactState =
-                      es
-                        & ensPrevGovActionIdsChildrenL .~ updatePrevGovActionIdsChildren gas ensPrevGovActionIdsChildren
-                  , rsEnacted = Set.insert gasId rsEnacted
-                  , rsRemoved = Set.union (Set.delete gasId siblings) rsRemoved
-                  , rsDelayed = delayingAction gasAction
-                  }
+          let
+            st' =
+              st
+                & rsEnactStateL .~ newEnactState
+                & rsDelayedL .~ delayingAction gasAction
+                & rsEnactedL %~ (Seq.:|> gasId)
```

Comment:
Switch this to the actual `GovernaceActionState`. Many things will become simpler.
```suggestion
                & rsEnactedL %~ (Seq.:|> gas)
```